Implementering av Admin side for Gi Din Stemme

junus-sefa opened the pull request.

Commit:
First commit

Commit:
Added

Commit:
Added idle modal + fix

- Added Idle modal
- Changed som Audio settings in completedControlsSmall.jsx

Commit:
++

Commit:
Big commit

Commit:
Big commit

Commit:
Small commit

Commit:
Big commit (some structural changes)

Commit:
Commit

tonystr committed:
feat: add autoprefixer dependency

Commit:
Last commit før merge

Commit:
Merge branch 'master' into Dashboard

Commit:
Fix

Commit:
Added play/pause button to contribution page

Commit:
Changed styling of input field on table

Commit:
Lagt til "visnings"-side for single texts

Commit:
update

tonystr commented on src/dashboard/components/chart/ChartUserAge.jsx:
```diff
@@ -0,0 +1,91 @@
+            </div>
+        );
+    }
+}
```
Vi vil helst unngå å bruke class-based components med mindre det er veldig vanskelig å unngå. Her er det lett å konvertere til function component:

```jsx
function ChartUserAge() {
    const options = {...};
    const series = [...];
    
    return (
            <div className="app">
                <div className="row">
                    <div className="mixed-chart">
                        <Chart
                            options={options}
                            series={series}
                            type="polarArea"
                            height="450px"
                            width="100%"
                        />
                    </div>
                </div>
            </div>
    );
}
```

Det er heller ingen grunn for å large series og options i state, når dataen aldri skal endres inne i dette Componentet. I fremtiden kommmer vi til å enten passe dataen inn gjennom props, eller gjøre api requests istedenfor å definere dataen her.

junus-sefa replied:
Done

tonystr commented on src/dashboard/components/chart/ChartUserDialect.jsx:
```diff
@@ -0,0 +1,95 @@
+import React, { Component } from 'react';
+import Chart from 'react-apexcharts';
+
+class ChartUserDialect extends Component {
```
Bruk function component her også

junus-sefa replied:
Done

tonystr commented on src/dashboard/components/chart/ChartUserGender.jsx:
```diff
@@ -0,0 +1,89 @@
+import React, { Component } from 'react';
+import Chart from 'react-apexcharts';
+
+class ChartUserGender extends React.Component {
```
Bruk function component her også

junus-sefa replied:
Done

tonystr commented on src/dashboard/components/chart/ChartUsersPerDay.jsx:
```diff
@@ -0,0 +1,136 @@
+import React, { Component } from 'react';
+import Chart from 'react-apexcharts';
+
+class ChartUsersPerDay extends Component {
```
Bruk function component her også

tonystr commented on src/dashboard/components/shared/DashMenu.jsx:
```diff
@@ -0,0 +1,90 @@
+        {
+            title: 'Logg ut',
+            icon: ArrowLeftOnRectangleIcon,
+            path: '/gdsadmin/',
```
kan vi rename til bare admin?

junus-sefa replied:
Admin er et generelt navn alle vet om. Så for sikkerhetsgrunner er denne endret til noe litt utenom det normale.

tonystr commented on src/dashboard/components/shared/DashMenu.jsx:
```diff
@@ -0,0 +1,90 @@
+        >
+            <LeftArrow
+                className={`bg-fred h-12 w-12 text-natt rounded-full hover:rotate-12 transition p-1 absolute -right-5 top-[6.5rem] cursor-pointer ${
+                    !open && 'rotate-180'
```
dette burde være en ternary som returnerer en tom streng hvis open er true, ellers vil den legge til en klasse som heter "false"

tonystr commented on src/dashboard/components/shared/DashMenu.jsx:
```diff
@@ -0,0 +1,90 @@
+                />
+                <h1
+                    className={`text-natt origin-left font-medium text-2xl ${
+                        !open && 'scale-0'
```
bruk ternary her også

tonystr commented on src/dashboard/components/shared/DashMenu.jsx:
```diff
@@ -0,0 +1,90 @@
+                            <div className="text2xl block w-6">
+                                <item.icon />
+                            </div>
+                            <div className={`font-fet ${!open && 'hidden'}`}>
```
bruk ternary her også

tonystr commented on src/dashboard/components/table/ContributionTable.jsx:
```diff
@@ -0,0 +1,374 @@
+export const ContributionTable = ({ columns, rows, tableId }) => {
+    //
+    const [playBtn, setPlayBtn] = useState(true);
+    //
```
tomme kommentarer?

tonystr commented on src/dashboard/components/table/ContributionTable.jsx:
```diff
@@ -0,0 +1,374 @@
+                                    return (
+                                        <>
+                                            <ArrowsUpDownIcon className="h-5 w-5" />
+                                        </>
```
hvorfor bruker vi <></> her? Det virker hennsiktsløst, kan få mer leselig kode med å fjerne dem og paranteser (altså bare `return <Component className="h-5 w-5" />;`)

tonystr commented on src/dashboard/components/table/StandardTable.jsx:
```diff
@@ -0,0 +1,526 @@
+                                    return (
+                                        <>
+                                            <ArrowsUpDownIcon className="h-5 w-5" />
+                                        </>
```
samme kommentar om  <></>

junus-sefa replied:
Done

tonystr commented on src/dashboard/components/table/ContributionTable.jsx:
```diff
@@ -0,0 +1,374 @@
+                                                            }
+                                                            onChange={(event) =>
+                                                                handleSearch(
+                                                                    event.target
```
bruk `e` istedenfor order `event`. `event` er et reservert nøkkelord i javascript, det kan lede til problemer

tonystr reviewed (changes requested):
Har ikke sett igjennom alle filene, kommer sannsynligvis med flere kommentarer imorgen

Commit:
Fixes

Commit:
Graf riktig demo data

tonystr commented on src/dashboard/components/table/StandardTable.jsx:
```diff
@@ -0,0 +1,490 @@
+    EyeIcon,
+} from '@heroicons/react/24/solid';
+//Toast Library
+import toast, { Toaster } from 'react-hot-toast';
```
trenger ikke å importere Toaster her

tonystr commented on src/dashboard/pages/settings/Settings.jsx:
```diff
@@ -0,0 +1,134 @@
+    ArrowTopRightOnSquareIcon,
+} from '@heroicons/react/24/outline';
+//Toast Library
+import toast, { Toaster } from 'react-hot-toast';
```
trenger ikke å importere Toaster her

tonystr commented on src/dashboard/components/table/StandardTable.jsx:
```diff
@@ -0,0 +1,490 @@
+        setSort({ order: 'asc', orderBy: 'id' });
+        setActivePage(1);
+        setFilters({});
+        toast.success('Nullstilling vellykket!');
```
Dette er ikke en handling som kan feile, så burde bare være `toast('Nullstillt')`

tonystr commented on src/dashboard/components/table/addon/TableHelpers.jsx:
```diff
@@ -0,0 +1,111 @@
+/* eslint-disable eqeqeq */
+
```
denne filen exporterer ikke et react komponent som default, så den burde ikke være navngitt som et komponent. Endre til `tableHelpers.js`

tonystr commented on src/dashboard/components/table/addon/TablePagination.jsx:
```diff
@@ -0,0 +1,80 @@
+    ChevronRightIcon,
+} from '@heroicons/react/24/solid';
+
+export const Pagination = ({
```
denne burde exporteres som default function, ikke const: `export default function Pagination({` (husk å fjerne => på linje 16)

junus-sefa replied:
Går ikke å eksportere denne som en function.

tonystr replied:
Hvorfor ikke? Husk at når noe eksporteres som default importeres det slik `import Pagination from...` istedenfor slik `import { Pagination } from...`. Filen burde forresten hete det samme som komponentet

junus-sefa replied:
![CleanShot 2023-03-02 at 13 35 41@2x](https://user-images.githubusercontent.com/72740756/222430699-af4241d0-a66e-4105-97a3-c495bd4892f8.jpg)
Får opp denne feilmeldingen.

tonystr commented on src/dashboard/components/table/addon/TablePagination.jsx:
```diff
@@ -0,0 +1,80 @@
+/* eslint-disable jsx-a11y/accessible-emoji */
```
Hvorfor trengs denne kommentaren? Jeg ser ingen emojis i filen. Er det for heroicons?

tonystr commented on src/dashboard/pages/tags/TagCategoriesPage.jsx:
```diff
@@ -0,0 +1,108 @@
+                                Legg til, slett eller endre de ulike dialektene.
+                            </p>
+                        </div>
+                    </NavLink>
```
Ikke bruk NavLink her, bare Link. NavLink er for lenker som vises likt på forskjellige sider

tonystr commented on src/dashboard/pages/tags/TagSingleCategoryPage.jsx:
```diff
@@ -0,0 +1,195 @@
+    const navigate = useNavigate();
+    //Location - used to determine if the user is creating a new tag or editing an existing one
+    const location = useLocation();
+    const categoryType = location.state;
```
her er det best om vi bruker `categoryType = location.state.categoryType;` og setter `state={{ categoryType: value }}` hvor enn det blir satt istedenfor `state={value}`

tonystr commented on src/dashboard/pages/tags/TagSinglePage.jsx:
```diff
@@ -0,0 +1,155 @@
+        } else {
+            navigate(`/gdsadmin/tags`);
+        }
+    }, [actionType]);
```
her må categoryType også være en dependency for effekten

tonystr commented on src/dashboard/pages/texts/TextSinglePage.jsx:
```diff
@@ -0,0 +1,372 @@
+    };
+    useEffect(() => {
+        countWords();
+    });
```
Hvorfor bruker vi en ref her? countWords kalles allerede av onChange. Virker ikke som useEffect eller ref er nødvendig her

junus-sefa replied:
Ref brukes for å hente input fra inputfelt. useEffect brukes for å oppdatere wordcount når bruker skriver i tekstfelt.

tonystr replied:
Input kan hentes med e.target.value i onChange, da slipper vi å bruke ref. Da trenger vi heller ikke useEffect

Commit:
Current Fixes

Commit:
Fix

Commit:
Fix

Commit:
TablePagination Fix

Commit:
NEW: Keywords, ADD: Viewpage for Texts, Targets and Keywords

- Target pages compromised
- ADD: Keyword pages + added in text-edit/new-page
- New action type on standardtable: editAndDeleteNoView
- Added keywords on Contribution table
- Added publish settings on tagSinglePage
- ADD: some styling of scrollbar in dashboard.css

tonystr reviewed (approved):
Veldig bra jobbet!
